lib/src/binary_codec.dart

```diff
@@ -199,12 +204,99 @@ class PostgresBinaryEncoder extends Converter<dynamic, Uint8List?> {
           }
           return outBuffer;
         }
+
+      case PostgreSQLDataType.point:
+        {
+          if (value is LatLng) {
+            final bd = ByteData(16);
+            bd.setFloat64(0, value.latitude);
+            bd.setFloat64(8, value.longitude);
+            return bd.buffer.asUint8List();
+          }
+          throw FormatException(
+              'Invalid type for parameter value. Expected: LatLng Got: ${value.runtimeType}');
+        }
+
+      case PostgreSQLDataType.integerArray:
+        {
+          if (value is List<int>) {
+            return writeListBytes<int>(value, 23, (_) => 4,
+                (bd, offset, item) => bd.setInt32(offset, item));
+          }
+          throw FormatException(
+              'Invalid type for parameter value. Expected: List<int> Got: ${value.runtimeType}');
+        }
+
+      case PostgreSQLDataType.textArray:
+        {
+          if (value is List<String>) {
+            final bytesArray = value.map((v) => castBytes(utf8.encode(v)));
+            return writeListBytes<Uint8List>(
+                bytesArray, 25, (item) => item.length, (bd, offset, item) {
+              item.forEach((i) => bd.setUint8(offset++, i));
+            });
+          }
+          throw FormatException(
+              'Invalid type for parameter value. Expected: List<String> Got: ${value.runtimeType}');
+        }
+
+      case PostgreSQLDataType.doubleArray:
+        {
+          if (value is List<double>) {
+            return writeListBytes<double>(value, 701, (_) => 8,
+                (bd, offset, item) => bd.setFloat64(offset, item));
+          }
+          throw FormatException(
+              'Invalid type for parameter value. Expected: List<double> Got: ${value.runtimeType}');
+        }
+
+      case PostgreSQLDataType.jsonbArray:
+        {
+          if (value is List<Object>) {
+            final bytesArray = value.map((v) {
+              final jsonBytes = utf8.encode(json.encode(v));
+              final writer = ByteDataWriter(bufferLength: jsonBytes.length + 1);
+              writer.writeUint8(1);
+              writer.write(jsonBytes);
+              return writer.toBytes();
+            });
+            return writeListBytes<Uint8List>(
+                bytesArray, 3802, (item) => item.length, (bd, offset, item) {
+              item.forEach((i) => bd.setUint8(offset++, i));
+            });
+          }
+          throw FormatException(
+              'Invalid type for parameter value. Expected: List<Object> Got: ${value.runtimeType}');
+        }
+        
       default:
         throw PostgreSQLException('Unsupported datatype');
     }
   }
 }
 
+Uint8List writeListBytes<T>(
```

Comment:
The rest of the package uses `package:buffer`'s `ByteDataWriter` to achieve things like this. Please use that, it would make the code more consistent.
https://pub.dev/documentation/buffer/latest/buffer/ByteDataWriter-class.html

Reply:
Done